tests/test_examples.py

```diff
@@ -81,37 +82,29 @@ def test_notnan(m):
     (lambda t: -t, "f(t) = -t", lambda t: -np.ones_like(t), "lin-neg"),
     (lambda t: t ** 2 - t + np.ones_like(t), "f(t) = t^2-t+1", lambda t: 2 * t -np.ones_like(t), "polynomial"),
     (lambda t: np.sin(t) + np.ones_like(t) / 2, "f(t) = sin(t)+1/2", lambda t: np.cos(t), "trig"),
-    (
-        lambda t: np.array([2 * t, - t]),
-        "f(t) = [2t, -t]",
-        lambda t: np.vstack((2 * np.ones_like(t), -np.ones_like(t))),
-        "2D linear",
-    ),
-    (
-        lambda t: np.array([np.sin(t), np.cos(t)]),
-        "f(t) = [sin(t), cos(t)]",
-        lambda t: np.vstack((np.cos(t), -np.sin(t))),
-        "2D trig",
-    ),
+    (lambda t: np.array([2 * t, - t]), "f(t) = [2t, -t]", lambda t: np.vstack((2 * np.ones_like(t), -np.ones_like(t))), "2D linear"),
+    (lambda t: np.array([np.sin(t), np.cos(t)]), "f(t) = [sin(t), cos(t)]", lambda t: np.vstack((np.cos(t), -np.sin(t))),  "2D trig"),
 )
+@pytest.mark.filterwarnings('ignore::sklearn.exceptions.ConvergenceWarning') # some methods can throw these for examples
 @pytest.mark.parametrize("m", methods)
-@pytest.mark.parametrize("func_spec", funcs_and_derivs, ids=(tup[-1] for tup in funcs_and_derivs))
+@pytest.mark.parametrize("func_spec", funcs_and_derivs)
 def test_fn(m, func_spec):
     func, fname, deriv, f_id = func_spec
-    t = np.linspace(0, 2*np.pi, 100)
-    if m == 'trend_filtered':
-        # Add noise to avoid all zeros non-convergence warning for sklearn lasso
-        f_mod = lambda t: func(t) + 1e-9 * np.random.randn(*t.shape) # rename to avoid infinite loop
+    t = np.linspace(0, 2*np.pi, 100, endpoint=False) # For periodic functions, it's important the endpoint not be included
+    if m == "spectral":
```

Comment:
I decided to treat spectral as a special case in here, because it already kind of was, and this was the easiest extension to get coverage.